src/main/scala/io/qbeast/core/model/DeleteFileBuilder.scala

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2021 Qbeast Analytics, S.L.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.qbeast.core.model
+
+/**
+ * Builder for creating IndexFile instances.
+ */
+final class DeleteFileBuilder {
```

Comment:
Do we need a FileBuilder? It is not really a pattern in Scala, and we are using it for the IndexFiles only because we don't have all the block information when we create the file at the beginning. But if it is not the case with DeleteFiles (meaning, when we make a DeleteFile, we have path, size, and deletionTimestamp), I would avoid it.